### PR TITLE
fix: Disconnect MongoDB client before reconnecting

### PR DESCRIPTION
**Pull request by odubajDT**
Signed-off-by: odubajDT <ondrej.dubaj@dynatrace.com>

## This PR

- disconnect mongoDB client before reconnecting

### Integration tests
https://github.com/keptn/keptn/actions/runs/2152967134

### CODE REVIEW

File: mongodb-datastore/db/mongodb_connection.go [hunk 1]
```diff
@@ -17,6 +19,9 @@ var mongoDBConnectionInstance *MongoDBConnection
 
 var mongoConnectionOnce sync.Once
 
+const clientCreationFailed = "failed to create mongo client: %v"
+const clientConnectionFailed = "failed to create mongo client: %v"
```
Comment (thisthat) [inline mongodb-datastore/db/mongodb_connection.go:23]: why do we need 2 variables for this?

File: mongodb-datastore/db/mongodb_connection.go [hunk 2]
```diff
@@ -50,35 +55,40 @@ func (m *MongoDBConnection) EnsureDBConnection() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	if m.client == nil {
-		fmt.Println("No MongoDB client has been initialized yet. Creating a new one.")
+		logger.Info("No MongoDB client has been initialized yet. Creating a new one.")
 		return m.connectMongoDBClient()
 	} else if err = m.client.Ping(ctx, nil); err != nil {
-		fmt.Println("MongoDB client lost connection. Attempt reconnect.")
+		logger.Info("MongoDB client lost connection. Attempt reconnect.")
```
Comment (mowies) [inline mongodb-datastore/db/mongodb_connection.go:61]: For consistency reasons, I would reword this to "Attempting reconnect"
Like above with "Creating a new one"

Comment (odubajDT) [reply mongodb-datastore/db/mongodb_connection.go:61]: sure, will edit in all PRs, thx :)

**Review by thisthat — DISMISSED** [review]

**Review by bacherfl — APPROVED** [review]
